openwisp_radius/api/freeradius_views.py

```diff
@@ -300,43 +300,99 @@ def get_replies(self, user, organization_id):
             for reply in self.get_group_replies(user_group.group):
                 data.update({reply.attribute: {"op": reply.op, "value": reply.value}})
 
-            group_checks = get_group_checks(user_group.group)
-
-            for Counter in app_settings.COUNTERS:
-                group_check = group_checks.get(Counter.check_name)
-                if not group_check:
-                    continue
-                try:
-                    counter = Counter(
-                        user=user, group=user_group.group, group_check=group_check
-                    )
-                    remaining = counter.check()
-                except SkipCheck:
-                    continue
-                # if max is reached send access rejected + reply message
-                except MaxQuotaReached as max_quota:
-                    data.update(self.reject_attributes.copy())
-                    if "Reply-Message" not in data:
-                        data["Reply-Message"] = max_quota.reply_message
-                    return data, self.max_quota_status
-                # avoid crashing on unexpected runtime errors
-                except Exception as e:
-                    logger.exception(f'Got exception "{e}" while executing {counter}')
-                    continue
-                if remaining is None:
-                    continue
-                reply_name = counter.reply_name
-                # send remaining value in RADIUS reply, if needed.
-                # This emulates the implementation of sqlcounter in freeradius
-                # which sends the reply message only if the value is smaller
-                # than what was defined to a previous reply message
-                if reply_name not in data or remaining < self._get_reply_value(
-                    data, counter
-                ):
-                    data[reply_name] = remaining
+            group_checks = get_group_checks(
+                user_group.group, include_additional_checks=True
+            )
+
+            # Validate additional radius checks
+            additional_checks_result = self._validate_additional_radius_checks(
```

Comment:
Why don't we just execute the simultaneous check directly here? Without additional logic? I still don't see the benefits of this "additional radius checks" thing. It reminded me of my early errors in the infant openwisp2 of trying to do too many things too early :joy: :stuck_out_tongue:, I am still traumatized.

When in the future we'll need to support more checks, we can just update the code.